pkg/providers/vsphere/virtualmachine/snapshot.go

```diff
@@ -188,25 +188,23 @@ func (m snapshotMap) add(parent string, tree []vimtypes.VirtualMachineSnapshotTr
 	}
 }
 
-// GetSnapshotSize calculates the size of a given snapshot in bytes. It includes
-// the memory file, the vmdk files, and the vmsn file, and excludes FCDs.
-// The algorithm use almost the same logic as how VC UI calculates the snapshot size.
-// The only difference is that this function does not include the size of FCDs,
-// and returns an error if the snapshot is not found.
-func GetSnapshotSize(logger logr.Logger,
+// GetSnapshotSize calculates the size of a given snapshot in bytes. It returns
+// two sizes, first one includes the memory file, the vmdk files, and the vmsn
+// file, excludes FCDs. Second one only includes FCDs.
+// The algorithm use almost the same logic as how VC UI calculates the snapshot
+// size. The only difference is that this function returns the size of FCDs
+// as a different variable, and returns an error if the snapshot is not found.
+func GetSnapshotSize(
+	logger logr.Logger,
 	moVM mo.VirtualMachine,
-	vmSnapshot *vimtypes.ManagedObjectReference) (int64, error) {
+	vmSnapshot *vimtypes.ManagedObjectReference) (int64, int64, error) {
 	if vmSnapshot == nil || vmSnapshot.Value == "" {
-		return 0, fmt.Errorf("moSnapshot is nil or empty")
-	}
-
-	if moVM.Config == nil {
-		return 0, fmt.Errorf("moVM.Config is nil")
+		return 0, 0, fmt.Errorf("moSnapshot is nil or empty")
```

Comment:
If the error is a constant string, use `errors.New`.  That or consider defining sentinel errors.